x-pack/plugin/esql-core/src/test/java/org/elasticsearch/xpack/esql/core/expression/function/scalar/FunctionTestUtils.java

```diff
@@ -15,10 +16,13 @@
 public final class FunctionTestUtils {
 
     public static Literal l(Object value) {
-        return new Literal(EMPTY, value, DataType.fromJava(value));
+        return l(value, DataType.fromJava(value));
     }
 
     public static Literal l(Object value, DataType type) {
+        if (value instanceof String && (type == DataType.TEXT || type == DataType.KEYWORD)) {
```

Comment:
NIT: `type == DataType.TEXT || type == DataType.KEYWORD` theck looks a bit cheaper. Do you mind making it first (before instanceof)?